lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1580,8 +1580,8 @@ def cmd_set(*args)
     # Set PAYLOAD
     if name.upcase == 'PAYLOAD' && active_module && (active_module.exploit? || active_module.evasion?)
       if value.start_with?('/', 'payload/')
-        # Trims starting `/`, `payload/`, `/payload/` from user input
-        value.sub!(%r{^/?(?:payload/)?}, '')
+        # Trims extraneous information from a full path to a file
+        value = trim_path(value, "payload")
```

Comment:
Can we be consistent about single quotes or double quotes?

Reply:
It seems that you're a double quotes guy, which is totally fine. I do that in Python. You use an empty single-quoted string earlier, so I was wondering if there's a particular style you were going for.

Reply:
Googled about and got a good description for when to use what quotes I was lacking before

- `"`When you're using string interpolation
- `'` The rest of the time

I've got that in my head so now I'll have a reason to use the right quote at the right time as opposed to whatever quote was the most popular in my brain when I was typing 🙃

Reply:
It's up to you! Here's the rule we generally follow: https://github.com/rubocop-hq/ruby-style-guide#consistent-string-literals. Folks can choose either approach. I used to do double quotes but switched to single. It only matters to be consistent. :)

Reply:
I just wasn't sure if you were preferring single-quoted string literals and double-quoted _empty_ string literals.